t/lib/NoFork.pm

```diff
@@ -4,7 +4,21 @@ BEGIN {
     *CORE::GLOBAL::fork = sub { die "you should not fork" };
 }
 use Config;
-tied(%Config)->{d_fork} = 0;    # blatant lie
+if ($Config::Config{d_fork}) {
```

Comment:
%Config is readonly in theory but in apparently one of my predecessors found an ugly way around that. XSConfig is more strictly readonly.

A far cleaner solution IMO to this would be to make a writable copy of `%Config`, modify the `d_fork` entry and replace the hash member in *Config::Config with this new hash (or untie and assign). This would need to be done before anything else imports Config.

~~~~
perl  -E 'require Config; my %conf = %Config::Config; $conf{cc} = "something"; *Config::Config = \%conf; Config->import; say $Config::Config{cc}'
~~~~